### PR TITLE
I18n Implementation on Company Update Form

### PR DESCRIPTION
**Pull request by Shinsina**
Screen Shot 2021-05-21 at 3 34 35 PM
Screen Shot 2021-05-21 at 3 38 18 PM
Screen Shot 2021-05-21 at 3 38 15 PM
Screen Shot 2021-05-21 at 3 38 10 PM
Screen Shot 2021-05-21 at 3 36 15 PM
Screen Shot 2021-05-21 at 3 35 17 PM
Screen Shot 2021-05-21 at 3 35 12 PM
Screen Shot 2021-05-21 at 3 34 57 PM
Screen Shot 2021-05-21 at 3 34 48 PM

### CODE REVIEW

**Comment by Shinsina** [comment]
[How to solve for Mundo Leaders entries in Spanish and English](https://trello.com/c/oKzezqPm/50-how-to-solve-for-mundo-leaders-entries-in-spanish-and-english)

**Comment by Shinsina** [comment]
This currently doesn't work on the review form YET, I need to update more graphql queries/fragments in order to resolve that.

**Comment by Shinsina** [comment]
This should be all the forms updated correctly.

File: services/app/app/gql/queries/config.graphql [hunk 1]
```diff
@@ -8,5 +8,6 @@ query CompanyUpdateConfig {
     leadershipCompanyLabel
     leadershipSectionAlias
     leadershipSectionMax
+    lang
```
Comment (solocommand) [inline services/app/app/gql/queries/config.graphql:11]: Call this `locale` please

File: docker-compose.yml [hunk 2]
```diff
@@ -69,6 +69,7 @@ services:
       LEADERSHIP_SECTION_MAX: ${LEADERSHIP_SECTION_MAX-3}
       CONTACT_URL: ${CONTACT_URL}
       CONTACT_TEXT: ${CONTACT_TEXT-Contact Us}
+      APPLICATION_LANG_CHANGE: ${APPLICATION_LANG_CHANGE}
```
Comment (solocommand) [inline docker-compose.yml:72]: rename to `APP_LOCALE`

File: services/app/app/helpers/i18n.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+import translations from './translations';
+import { helper } from '@ember/component/helper';
+
+export function i18n ([lang, fieldLabelKey]) {
+  console.log(lang)
```
Comment (solocommand) [inline services/app/app/helpers/i18n.js:5]: Remove log

File: services/app/app/helpers/i18n.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import translations from './translations';
+import { helper } from '@ember/component/helper';
+
+export function i18n ([lang, fieldLabelKey]) {
```
Comment (solocommand) [inline services/app/app/helpers/i18n.js:4]: Don't pass in the language value, look it up from the configuration. This should use the [ember-intl](https://ember-intl.github.io/ember-intl/versions/master/docs/quickstart) package so we're not re-inventing everything here.

File: services/app/app/routes/application.js [hunk 5]
```diff
@@ -21,6 +21,7 @@ export default Route.extend(ApplicationRouteMixin, {
     leadershipCompanyLabel,
     leadershipSectionAlias,
     leadershipSectionMax,
+    lang
```
Comment (solocommand) [inline services/app/app/routes/application.js:24]: Make these `locale` instead of `lang`.

File: services/app/app/templates/review/company.hbs [hunk 6]
```diff
@@ -15,36 +15,42 @@
       submission=model.submission
       toggleField=(action "toggleField")
       payload=payload
+      lang=config.lang
```
Comment (solocommand) [inline services/app/app/templates/review/company.hbs:18]: None of this should be passed through anymore

File: services/graphql/src/env.js [hunk 7]
```diff
@@ -41,4 +41,5 @@ module.exports = cleanEnv(process.env, {
   LEADERSHIP_SECTION_MAX: num({ desc: 'The maximum number of leadership sections that can be selected per site.', default: 3 }),
   CONTACT_URL: str({ desc: 'If configured, the URL that will be added to the navigation for support requests.', default: '' }),
   CONTACT_TEXT: str({ desc: 'Link text for navigation element', default: 'Contact Us' }),
+  APPLICATION_LANG_CHANGE: str({ desc: 'If the language needs to be changed to something besides English', default: 'en' }),
```
Comment (solocommand) [inline services/graphql/src/env.js:44]:
```suggestion
  APP_LOCALE: str({ desc: 'The application locale', choices: ['en-us', 'es-mx'], default: 'en-us' }),
```

**Review by solocommand — CHANGES_REQUESTED** [review]
A few changes needed, notably using the ember-intl addon instead of a custom helper.

File: services/graphql/src/env.js (hunk 7 above)
Comment (Shinsina) [reply services/graphql/src/env.js:44]: Screen Shot 2021-05-24 at 12 51 13 PM

**Comment by Shinsina** [comment]
> A few changes needed, notably using the ember-intl addon instead of a custom helper.

Screen Shot 2021-05-24 at 12 51 13 PM

**Pull request merged**